crates/router/src/connector/opennode/transformers.rs

```diff
@@ -21,10 +52,12 @@ pub struct OpennodePaymentsRequest {
     callback_url: String,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for OpennodePaymentsRequest {
+impl TryFrom<&OpennodeRouterData<&types::PaymentsAuthorizeRouterData>> for OpennodePaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        get_crypto_specific_payment_data(item)
+    fn try_from(
+        item: &OpennodeRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
+        get_crypto_specific_payment_data(item.router_data)
```

Comment:
Pass the complete `item` Object to this function

```suggestion
        get_crypto_specific_payment_data(item)
```

Also in the function definition of ` get_crypto_specific_payment_data` change the argument from `&types::PaymentsAuthorizeRouterData` to `&OpennodeRouterData<&types::PaymentsAuthorizeRouterData>`

And there use `let amount = item.amount;` directly

Reply:
@SamraatBansal I have made the necessary modifications. 
Please let me know if I need to change anything else.